Add more Rails integration apps

delner opened the pull request:
## Purpose

To improve our integration test coverage, and to have sandbox environments to test/debug/experiment with `ddtrace` behavior.

## What's changed

 - Some fixes to Rails 5 application
 - Added a Rails 6 application
 - Added a Rails 7 application

Additionally...

 - Enabled more features in Rails integration apps by default (tracing, profiling, logs, runtime metrics)
 - Bumped Ruby 3.1 image (for integration apps) from `preview1` to `latest`.
 - Some tweaks to `DemoEnv`
 - More detailed documentation in Rails integration apps
 - Added Rails 6, 7 to integration CI tests.

ivoanjo commented on integration/apps/rails-six/script/build-images:
```diff
@@ -0,0 +1,34 @@
+  docker build --build-arg BASE_IMAGE=datadog/dd-apm-demo:rb-2.6 -t datadog/dd-apm-demo:rb-2.6-rails-six .
+  docker build --build-arg BASE_IMAGE=datadog/dd-apm-demo:rb-2.7 -t datadog/dd-apm-demo:rb-2.7-rails-six .
+  docker build --build-arg BASE_IMAGE=datadog/dd-apm-demo:rb-3.0 -t datadog/dd-apm-demo:rb-3.0-rails-six .
+  docker build --build-arg BASE_IMAGE=datadog/dd-apm-demo:rb-3.1 -t datadog/dd-apm-demo:rb-3.1-rails-six .
+fi
```
Minor: Same note about adding `# ADD NEW RUBIES HERE` as for the Rails 7 app :)

ivoanjo commented on integration/images/ruby/3.1/Dockerfile:
```diff
@@ -1,5 +1,4 @@
-# https://hub.docker.com/layers/ruby/library/ruby/3.1.0-preview1-bullseye/images/sha256-19f9abe757cd2a99262aded1177cedba0613cefc3c6181ce670b2259270a59d7
-FROM ruby@sha256:19f9abe757cd2a99262aded1177cedba0613cefc3c6181ce670b2259270a59d7
+FROM ruby:3.1
```
Ooops, missed updating this one. Good catch!

ivoanjo reviewed (approved):
👍  LGTM

lloeki reviewed (approved):
LGTM. I added the possibility to have AppSec to the mix using an `appsec` feature.

We may want to consider adding Sinatra to the mix too.

delner commented:
Addressed feedback.

In a different set of PRs I'd like to add these as apps as well:

  - Sinatra (Classic)
  - Sinatra (Modular)
  - Sidekiq
  - Resque
